src/main/java/org/numenta/nupic/network/Layer.java

```diff
@@ -1832,7 +1850,7 @@ public ManualInput call(ManualInput t1) {
                         
                         CLAClassifier c = (CLAClassifier)t1.getClassifiers().get(key);
                         ClassifierResult<Object> result = c.compute(
-                            recordNum, inputMap, t1.getSDR(), true, true);
+                            recordNum, inputMap, t1.getSDR(), isLearn, true);
```

Comment:
@antidata Thank you for bringing this up! Can you write this to the Hackers
ML ? I would like to discuss this openly because I'm wondering why "infer"
even exists?

On Tue, Sep 15, 2015 at 12:05 PM, antidata <notifications@github.com> wrote:

> In src/main/java/org/numenta/nupic/network/Layer.java
> <https://github.com/numenta/htm.java/pull/307#discussion_r39536419>:
>
> > @@ -1832,7 +1850,7 @@ public ManualInput call(ManualInput t1) {
> >
> >                          CLAClassifier c = (CLAClassifier)t1.getClassifiers().get(key);
> >                          ClassifierResult<Object> result = c.compute(
> > -                            recordNum, inputMap, t1.getSDR(), true, true);
> > +                            recordNum, inputMap, t1.getSDR(), isLearn, true);
>
> I like this! Question @cogmission <https://github.com/cogmission>, would
> it be useful to get rid of infer hardcoding like you did with isLearn ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/htm.java/pull/307/files#r39536419>.
>



-- 
*With kind regards,*

David Ray
Java Solutions Architect

*Cortical.io <http://cortical.io/>*
Sponsor of:  HTM.java <https://github.com/numenta/htm.java>

d.ray@cortical.io
http://cortical.io